template/v1/Dockerfile

```diff
@@ -74,6 +74,9 @@ RUN install-glue-kernels && \
     sed -i '/if not store_history and (/i\        if "sm_analytics_runtime_check" in code:\n            return await self._complete_cell()\n' \
     "$SITE_PACKAGES/aws_glue_interactive_sessions_kernel/glue_kernel_base/BaseKernel.py"
 
+# Install npm dependencies
+RUN npm install -g vscode-jsonrpc && \
+    npm install -g sql-language-server
```

Comment:
1. Do these NPM packages have any relation to other packages the image? For example, do other packages depend on these NPM packages? How do we ensure compatibility between the NPM packages and other packages?

2. What is our strategy for updating npm packages and catching future regressions?
    1.  Without version locking NPM packages to patch level, builds may not be reproducible as each build can pull a new version from NPM.
    2. Without at least constraining lower and upper bounds for NPM package versions, we risk backwards incompatible changes in SageMaker Distribution patch/minor version releases, which won't follow SemVer guidelines.
    3. This Dockerfile is used for all v1 image versions. This could be problematic as we release new patch versions for old and new minor versions, as the version bumps for related packages--or the version bumps to these packages--may result in incompatibility (e.g., 1.1.x breaks but 1.2.x works). Should we start separating the Dockerfile out by minor and/or patch version, or declare NPM dependencies in another file which is separated out by minor and/or patch version?

4. Why are we installing via `npm`? Could we install these packages via conda/mamba instead? Are we planning to do that later?

5. Shouldn't this PR be adding tests for these NPM packages?

Reply:
1. Currently non of the packages will break if these are not installed. The newly added extension [PR](https://github.com/aws/sagemaker-distribution/pull/222/files) will use these to support sql auto-completion.
2. This was the guidance I got from the maintainers as currently there isn't a way to manage npm dependencies similar to conda-forge dependencies
3. these are not available in conda/mamaba, these are js only packages for JupyterServer.
4. Adding tests.